pkg/observability/modules/stackstate.ts

```diff
@@ -183,3 +194,49 @@ export function loadComponent(
 function token(apiToken: string, serviceToken: string): string {
   return apiToken ? `ApiToken ${ apiToken }` : `ApiKey ${ serviceToken }`;
 }
+
+/**
+ * Ensure that there is a NodeDriver called 'stackstate' that has the URL whitelisted, so that the
+ * Metadata Proxy can call out to the StackState API.
+ *
+ * NOTE: Be aware that this goes through Norman APIs, not the Steve ones.
+ *
+ * @param store
+ * @param url
+ * @returns
+ */
+export async function ensureObservabilityUrlWhitelisted(store: any, url: string): Promise<boolean> {
+  const nodeDrivers = await store.dispatch('rancher/findAll', { type: 'nodeDriver' }, { root: true });
+
+  let stackStateDriver;
+
+  if (!nodeDrivers || isEmpty(nodeDrivers)) {
+    // Need to create new
+  } else {
+    stackStateDriver = nodeDrivers.find((driver: any) => driver.name === 'stackstate');
+  }
+
+  if (!stackStateDriver) {
+    const emptyDriver = {
+      name: `stackstate`,
+      type: 'nodeDriver',
+      url:  'local://',
+    };
+
+    stackStateDriver = await store.dispatch('rancher/create', emptyDriver);
+  }
+
+  if (!stackStateDriver.whitelistDomains) {
+    stackStateDriver.whitelistDomains = [];
```

Comment:
Is it okay to mutate a value from the store? will (should?) the UI reflect the changes after `.push()`?

Reply:
I suggest we shouldn't to do that 😛